mypy.ini

```diff
@@ -21,16 +22,14 @@ exclude = (?x)(
 # *.extern modules that actually live in *._vendor will also cause attr-defined issues on import
 disable_error_code = attr-defined
 
-# - Avoid raising issues when importing from "extern" modules, as those are added to path dynamically.
-#   https://github.com/pypa/setuptools/pull/3979#discussion_r1367968993
+[mypy-pkg_resources._vendor.*,setuptools._vendor.*,setuptools.config._validate_pyproject.*,setuptools.tests.*]
+# Even when excluded there might be problems: python/mypy/issues/11936#issuecomment-1466764006
+# - Avoid raising issues when importing from "_vendor" modules
+# - Avoid raising issues when importing dependencies in tests
```

Comment:
```suggestion
[mypy-pkg_resources._vendor.*,setuptools._vendor.*]
# Avoid raising import issues in vendored modules. Even when excluded there might be problems: python/mypy/issues/11936#issuecomment-1466764006
```
Do you still need to specify tests here? These are modules that should also be found in `exclude`, tests are not excluded. Originally I added it because of the dynamically imported created modules that are imported, but you `type: ignore`d those, so this shouldn't be needed.

Similarly the `_validate_pyproject` issue was fixed upstream.

Also seems to work locally w/o.